pkg/auth/multiauthenticator.go

```diff
@@ -0,0 +1,81 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+)
+
+type IssuerWithClientID struct {
+	// Issuer of the token. The issuer cannot be empty.
+	Issuer string
+	// ClientID of the token. The clientID cannot be empty.
+	ClientID string
+}
+
+type multiAuthenticatorWithClientIDImpl struct {
+	authenticators map[IssuerWithClientID]Authenticator
+}
+
+func NewMultiAuthenticatorWithClientID(
+	authenticators map[IssuerWithClientID]Authenticator) (MultiAuthenticatorWithClientID, error) {
+	authMap := make(map[IssuerWithClientID]Authenticator)
+	for issuerWithClientID, authenticator := range authenticators {
+		if issuerWithClientID.ClientID == "" {
+			return nil, fmt.Errorf("clientID cannot be empty")
+		}
+		if issuerWithClientID.Issuer == "" {
+			return nil, fmt.Errorf("issuer cannot be empty")
+		}
+		authMap[issuerWithClientID] = authenticator
+	}
+	return &multiAuthenticatorWithClientIDImpl{
+		authenticators: authMap,
+	}, nil
+}
+
+func (m *multiAuthenticatorWithClientIDImpl) ListIssuersWithClientID() []IssuerWithClientID {
+	var issuerWithClientIDs []IssuerWithClientID
+	for issuerWithClientID, _ := range m.authenticators {
+		issuerWithClientIDs = append(issuerWithClientIDs, issuerWithClientID)
+	}
+	return issuerWithClientIDs
+}
+
+func (m *multiAuthenticatorWithClientIDImpl) GetAuthenticators(issuer string) []Authenticator {
+	var authenticators []Authenticator
+	for issuerWithClientID, authenticator := range m.authenticators {
+		if issuerWithClientID.Issuer == issuer {
+			authenticators = append(authenticators, authenticator)
+		}
+	}
+	return authenticators
+}
+
+func (m *multiAuthenticatorWithClientIDImpl) AuthenticateToken(ctx context.Context, accessToken string) (*Claims, error) {
```

Comment:
accessToken -> idToken ?

Reply:
I thought I changed it everywhere. My bad will fix this.